deepcell/image_generators/cropping.py

```diff
@@ -0,0 +1,493 @@
+                applied, for debugging purposes.
+            save_prefix (str): Prefix to use for saving sample
+                images (if save_to_dir is set).
+            save_format (str): Format to use for saving sample images
+                (if save_to_dir is set).
+
+        Returns:
+            SemanticIterator: An Iterator yielding tuples of (x, y),
+                where x is a numpy array of image data and y is list of
+                numpy arrays of transformed masks of the same shape.
+        """
+        return CroppingIterator(
+            train_dict,
+            self,
+            batch_size=batch_size,
+            transforms=transforms,
+            transforms_kwargs=transforms_kwargs,
+            shuffle=shuffle,
+            min_objects=min_objects,
+            seed=seed,
+            data_format=self.data_format,
+            save_to_dir=save_to_dir,
+            save_prefix=save_prefix,
+            save_format=save_format,
+            crop_size=self.crop_size)
+
+    def get_random_transform(self, img_shape, seed=None):
+        transform_parameters = \
+            super(CroppingDataGenerator, self).get_random_transform(img_shape=img_shape,
+                                                                    seed=seed)
+
+        crop_indices = None
+        if self.crop_size is not None:
+
+            img_dims = img_shape[1:] if self.channel_axis == 1 else img_shape[:2]
+            if img_dims == self.crop_size:
+                # don't need to crop
+                pass
+            elif img_dims[0] == self.crop_size[0] or img_dims[1] == self.crop_size[1]:
+                raise ValueError('crop_size must be a subset of both axes or exactly '
+                                 ' equal to image dims')
+            elif img_dims[0] < self.crop_size[0] or img_dims[1] < self.crop_size[1]:
+                raise ValueError('Crop dimensions must be smaller than image dimensions')
+            else:
+                row_start = np.random.randint(0, img_dims[0] - self.crop_size[0])
+                col_start = np.random.randint(0, img_dims[1] - self.crop_size[1])
+                crop_indices = ([row_start, row_start + self.crop_size[0]],
+                                [col_start, col_start + self.crop_size[1]])
+
+        transform_parameters['crop_indices'] = crop_indices
+
+        return transform_parameters
+
+    def apply_transform(self, x, transform_parameters):
+
+        if transform_parameters['crop_indices'] is not None:
+            row_indices, col_indices = transform_parameters['crop_indices']
+            if self.channel_axis == 1:
+                x = x[:, row_indices[0]:row_indices[1], col_indices[0]:col_indices[1]]
+            else:
+                x = x[row_indices[0]:row_indices[1], col_indices[0]:col_indices[1], :]
+
+        x = super(CroppingDataGenerator,
+                  self).apply_transform(x=x, transform_parameters=transform_parameters)
```

Comment:
```suggestion
        x = super(CroppingDataGenerator, self).apply_transform(
            x=x, transform_parameters=transform_parameters)
```

Same here, personal preference but I think this is a bit easier to read.